gem_is_master.rb

```diff
@@ -0,0 +1,67 @@
+require_relative "./github_api";
+
+def put_warnings(warnings, warnings_count)
+  if warnings_count > 0
+    warnings.sort_by!{|gem| gem["name"]}
+    warn "\nThe following invoca gems in the gemfile are pointing behind the tips of their master branches."
+    puts ""
+    warnings.each do |gem|
+      puts gem
+      puts ""
+    end
+  end
+end
+
+lock = Bundler.read_file(Bundler.default_lockfile)
+gems = Bundler::LockfileParser.new(lock)
+incorrect_references = []
+warnings = []
+count = 0
+warnings_count = 0
+
+# Loop over all gem sources within the Gemfile
+gems.sources.each do |source|
+  begin
+    if source.class == Bundler::Source::Git
+      name = source.name
+      sha = source.revision
+      username = ENV['GITHUB_USERNAME']
+      password = ENV['GITHUB_PASSWORD']
+      forked = GithubRepoApi.new(username, password, "Invoca/#{name}").information!["fork"]
+      next if forked
+      master_sha = GithubRepoApi.new(username, password, "Invoca/#{name}").master_information["sha"]
+      ahead_by = GithubRepoApi.new(username, password, "Invoca/#{name}").sha_compare!(master_sha, sha)["ahead_by"]
+      behind_by = GithubRepoApi.new(username, password, "Invoca/#{name}").sha_compare!(master_sha, sha)["behind_by"]
```

Comment:
You're instantiating four separate instances of `GithubRepoApi` here, all for the same repo.

It'd be best to instantiate it once at the top of your loop and reuse it as necessary.

For example:
```suggestion
      repo = GithubRepoApi.new(username, password, "Invoca/#{name}")
      next if repo.information!["fork"]
      master_sha = repo.master_information["sha"]
      ahead_by = repo.sha_compare!(master_sha, sha)["ahead_by"]
      behind_by = repo.sha_compare!(master_sha, sha)["behind_by"]
```

That said, there are several things going on here that might be easier to explain in person. Let's have a quick chat and I can run through some concerns regarding:
* Refactoring `GithubRepoApi` to provide more specific methods like `forked?`, `master_sha`, etc.
* Method naming conventions
* Caching data so you aren't hitting the remote API multiple times in a row
* Moving error handling closer to the parts of code that can trigger it